src/main/java/uk/gov/hmcts/divorce/divorcecase/validation/ValidationUtil.java

```diff
@@ -171,6 +171,16 @@ public static List<String> validateCaseFieldsForIssueApplication(MarriageDetails
         );
     }
 
+    public static List<String> validateCaseFieldsForPersonalAndSolicitorService(final Application application,
+                                                                                final boolean applicant1ConfidentialContactDetails,
+                                                                                final boolean applicant2ConfidentialContactDetails) {
+        final boolean check = (application.isPersonalServiceMethod() || application.isSolicitorServiceMethod())
+            && (applicant1ConfidentialContactDetails || applicant2ConfidentialContactDetails);
+        return check
+            ? singletonList("You may not select Solicitor Service or Personal Service if the respondent is confidential.")
```

Comment:
Is this check correct? The error only mentions the respondent but the boolean checks if either party is confidential